static/components/TaskDetails.jsx

```diff
@@ -114,15 +115,34 @@ var TaskDetails = React.createClass({
   },
 
   updateBuildLog(data) {
-    var frag = document.createDocumentFragment();
-
     // add each additional new line
-    data.text.split('\n').forEach((line) => {
-      var div = document.createElement('div');
-      div.className = 'line';
-      div.innerHTML = ansi_up.ansi_to_html(line);
-      frag.appendChild(div);
-    });
+    var frag       = document.getElementsByClassName('frag')[0] || document.createDocumentFragment();
+    frag.className = 'frag';
+
+    var text       = data.chunks
+    var objLength  = data.chunks.length
+
+
+    for(var i = 0; i < objLength; i++){
+      var timer    = new Date(data.chunks[i].date)
+      var timeMil  = timer.getTime()
+      //Multiple by 60000 to convert offset to milliseconds
+      var offset   = timer.getTimezoneOffset() * 60000
+      var timezone = timeMil - offset
+      var newDate  = new Date(timezone)
+
+      var div  = document.createElement('div');
+      var time = document.createElement('div');
+
+      div.className  = 'line';
+      time.className = 'time';
+
+      div.innerHTML  = ansi_up.ansi_to_html(data.chunks[i].text)
+      time.innerHTML = moment(newDate).parseZone().format("h:mm a")
```

Comment:
I'd probably assign this to `time.textContent`, not `time.innerHTML`, since there is no HTML. _Shouldn't_ matter though ...